lib/semantics/check-omp-structure.h

```diff
@@ -12,24 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// OpenMP clause validity check for directives
+// OpenMP structure validity check list
+//    1. invalid clauses on directive
+//    2. invalid repeated clauses on directive
+//    3. TODO: invalid nesting of regions
 
 #ifndef FORTRAN_SEMANTICS_CHECK_OMP_STRUCTURE_H_
 #define FORTRAN_SEMANTICS_CHECK_OMP_STRUCTURE_H_
 
 #include "semantics.h"
+#include "../common/enum-set.h"
 #include "../parser/parse-tree.h"
-#include <typeindex>
 
 namespace Fortran::semantics {
 
+ENUM_CLASS(OmpDirectiveEnum, PARALLEL, DO, SECTIONS, SECTION, SINGLE, WORKSHARE,
+    SIMD, DECLARE_SIMD, DO_SIMD, TASK, TASKLOOP, TASKLOOP_SIMD, TASKYIELD,
+    TARGET_DATA, TARGET_ENTER_DATA, TARGET_EXIT_DATA, TARGET, TARGET_UPDATE,
+    DECLARE_TARGET, TEAMS, DISTRIBUTE, DISTRIBUTE_SIMD, DISTRIBUTE_PARALLEL_DO,
+    DISTRIBUTE_PARALLEL_DO_SIMD, PARALLEL_DO, PARALLEL_SECTIONS,
+    PARALLEL_WORKSHARE, PARALLEL_DO_SIMD, TARGET_PARALLEL, TARGET_PARALLEL_DO,
+    TARGET_PARALLEL_DO_SIMD, TARGET_SIMD, TARGET_TEAMS, TEAMS_DISTRIBUTE,
+    TEAMS_DISTRIBUTE_SIMD, TARGET_TEAMS_DISTRIBUTE,
+    TARGET_TEAMS_DISTRIBUTE_SIMD, TEAMS_DISTRIBUTE_PARALLEL_DO,
+    TARGET_TEAMS_DISTRIBUTE_PARALLEL_DO, TEAMS_DISTRIBUTE_PARALLEL_DO_SIMD,
+    TARGET_TEAMS_DISTRIBUTE_PARALLEL_DO_SIMD, MASTER, CRITICAL, BARRIER,
+    TASKWAIT, TASKGROUP, ATOMIC, FLUSH, ORDERED, CANCEL, CANCELLATION_POINT,
+    THREADPRIVATE, DECLARE_REDUCTION)
+
+using OmpDirectiveEnumType =
+    common::EnumSet<OmpDirectiveEnum, OmpDirectiveEnum_enumSize>;
+
+ENUM_CLASS(OmpClauseEnum, DEFAULTMAP, INBRANCH, MERGEABLE, NOGROUP, NOTINBRANCH,
+    NOWAIT, UNTIED, COLLAPSE, COPYIN, COPYPRIVATE, DEVICE, DIST_SCHEDULE, FINAL,
+    FIRSTPRIVATE, FROM, GRAINSIZE, LASTPRIVATE, NUM_TASKS, NUM_TEAMS,
+    NUM_THREADS, ORDERED, PRIORITY, PRIVATE, SAFELEN, SHARED, SIMDLEN,
+    THREAD_LIMIT, TO, LINK, UNIFORM, USE_DEVICE_PTR, IS_DEVICE_PTR, ALIGNED,
+    DEFAULT, DEPEND, IF, LINEAR, MAP, PROC_BIND, REDUCTION, SCHEDULE)
+
+using OmpClauseEnumType =
+    common::EnumSet<OmpClauseEnum, OmpClauseEnum_enumSize>;
```

Comment:
Same comments regarding these names.

Reply:
With regards to @tskeith's previous comments, I prefer not to include information about the type of a name embedded in the name itself.  Then, if I decide to change a name's type, I don't need to change its name.

Reply:
Addressed in commit [cb89744](cb89744746186bcf120d8ca62a791e2778366b9b).